src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add: function (num) {
+      this.result += num;
+
+      return this;
+    },
+
+    subtract: function (num) {
+      this.result -= num;
+
+      return this;
+    },
+
+    multiply: function (num) {
+      this.result *= num;
+
+      return this;
+    },
+
+    divide: function (num) {
+      this.result /= num;
+
+      return this;
+    },
+
+    reset: function () {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate: function (callback, num) {
+      callback.call(this, num);
+
+      return this;
```

Comment:
Use method shorthand syntax instead of the `function` keyword for defining methods in an object.